fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -209,17 +206,544 @@ fn expand_input_param(
+            },
+        );
+        custom_structs.insert(
+            "CoolIndieGame".to_string(),
+            Property {
+                name: "unused".to_string(),
+                type_field: "struct RedStickySquare".to_string(),
+                components: None,
+            },
+        );
+        let mut custom_enums = HashMap::new();
+        custom_enums.insert(
+            "EntropyCirclesEnum".to_string(),
+            Property {
+                name: "unused".to_string(),
+                type_field: "enum ComingUpWithThings".to_string(),
+                components: None,
+            },
+        );
+        custom_enums.insert(
+            "BurgundyBeefEnum".to_string(),
+            Property {
+                name: "unused".to_string(),
+                type_field: "enum SomeFrenchCuisine".to_string(),
+                components: None,
+            },
+        );
+        let abi_parser = ABIParser::new();
+        let result = expand_function(&the_function, &abi_parser, &custom_enums, &custom_structs);
+        // Some more editing was required because it is not rustfmt-compatible (adding/removing parentheses or commas)
+        let expected = TokenStream::from_str(
+            r#"
+#[doc = "Calls the contract's `HelloWorld` (0x00000000b6425163) function"]
+pub fn HelloWorld(
+    &self,
+    bim_bap_struct: BikeHelmet,
+    burgundy_beef_enum: SomeFrenchCuisine
+) -> ContractCall<((bool , u64 ,) , (bool, u64 ,))> {
+    Contract::method_hash(
+        &self.fuel_client,
+        &self.compiled,
+        [0, 0, 0, 0, 182, 66, 81, 99],
+        &[
+            ParamType::Struct(vec![ParamType::Bool, ParamType::U64]),
+            ParamType::Enum([Bool , U64])] , 
+            &[bim_bap_struct.into_token(), burgundy_beef_enum.into_token(),]
+    )
+    .expect("method not found (this should never happen)")
+}
+        "#,
+        );
+        let expected = expected.unwrap().to_string();
+        assert_eq!(result.unwrap().to_string(), expected);
+    }
+
+    // --- expand_selector ---
+    #[test]
+    fn test_expand_selector() {
+        let result = expand_selector(Selector::default());
+        assert_eq!(result.to_string(), "[0 , 0 , 0 , 0 , 0 , 0 , 0 , 0]");
+        let result = expand_selector(Selector::from([1, 2, 3, 4, 5, 6, 7, 8]));
+        assert_eq!(result.to_string(), "[1 , 2 , 3 , 4 , 5 , 6 , 7 , 8]");
+    }
+
+    // --- expand_fn_outputs ---
+    #[test]
+    fn test_expand_fn_outputs_zero_one_arg() {
+        let result = expand_fn_outputs(&[]);
+        assert_eq!(result.unwrap().to_string(), "()");
+        let result = expand_fn_outputs(&[Property {
+            name: "unused".to_string(),
+            type_field: "bool".to_string(),
+            components: None,
+        }]);
+        assert_eq!(result.unwrap().to_string(), "bool");
+        let result = expand_fn_outputs(&[Property {
+            name: "unused".to_string(),
+            type_field: String::from("struct streaming_services"),
```

Comment:
IIRC we're strictly following Rust naming convention in Sway, so a struct shouldn't come as `streaming_services`, it should be `StreamingServices`. Regardless, I believe `fuels-rs` should support both and turn them (through the codegen) into `StreamingServices`. Just in case.

Reply:
My idea was to feed it `snake_case` and put into light the fact that it turns them into `CamlCase`

Reply:
Oh, I see, even better then! Good stuff 😄